File: leetcode/0015.3sum/threeSum_test.py
from threeSum import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
        ("Example-1", [-1, 0, 1, 2, -1, -4], [
            [-1, 0, 1],
            [-1, -1, 2]
        ]),
]

@pytest.mark.parametrize(
    "example, nums, ans", 
    deepcopy(TEST_SUITS)
)
def test(example, nums, ans):
    res = SOL.threeSum(nums)
    assert len(res) == len(ans)

    res_set = set([tuple(sorted(arr)) for arr in res])
    ans_set = set([tuple(sorted(arr)) for arr in ans])
    assert res_set - ans_set == set()

File: leetcode/0015.3sum/threeSum_test.go
package leetcode

import (
	"reflect"
	"testing"
)

func Test_RemoveDuplicatesFromSortedArray(t *testing.T) {
	testSuits := []struct {
		name string
		nums []int
		ans  [][]int
	}{
		{"Example 1", []int{-1, 0, 1, 2, -1, -4}, [][]int{[]int{-1, 0, 1}, []int{-1, -1, 2}}},
	}

	for _, test := range testSuits {
		got := threeSum(test.nums)

		if !reflect.DeepEqual(got, test.ans) {
			t.Errorf("[%v] expect '%v', got '%v' ", test.name, test.ans, got)
		}
	}
}
